Master-Game/World/Map.cs

```diff
@@ -39,5 +53,91 @@ public void ResetTiles()
                 }
             }
         }
+
+        public MapData LoadJson()
+        {
+            string fileName = "Jay01.json";
+            string filePath = Path.GetDirectoryName(Path.GetDirectoryName(System.IO.Directory.GetCurrentDirectory())) + "\\Assets\\Maps\\" + fileName;
+            FileInfo f = new FileInfo(fileName);
+            string fullName = f.Name;
+            using (StreamReader reader = new StreamReader(filePath))
+            {
+                string myJson = reader.ReadToEnd();
+                MapData mapData = JsonConvert.DeserializeObject<MapData>(myJson);
+                return mapData;
+            }
+        }
+
+        public class MapData
+        {
+            //Name
+            public string name { get; set; }
+            //Version
+            public string version { get; set; }
+            //row
+            public int rows { get; set; }
+            //col
+            public int cols { get; set; }
+            //Tiles
+            public Tiles [] tiles { get; set; }
+            //Entities
+            public Entities[] entities { get; set; }
+
+            public int GetRows()
+            {
+                return rows;
+            }
+        }
+
+        public class Tiles
+        {
+            [JsonProperty("type")]
+            public int Type;
+
+            [JsonProperty("x")]
+            public int xCor;
+
+            [JsonProperty("y")]
+            public int yCor;
+        }
+
+        public class Entities
+        {
+            [JsonProperty("type")]
+            public string Type;
+
+            [JsonProperty("name")]
+            public string Name;
+
+            [JsonProperty("healthPoints")]
+            public string HealthPoints;
+
+            [JsonProperty("x")]
+            public string xCor;
+
+            [JsonProperty("y")]
+            public string yCor;
+        }
+
+        private BaseTile GetTile(int typeNum)
```

Comment:
This is what we call a factory. Put this in a new class. Call the class TileFactory. Make it a static public function.

Call the function CreateTile which takes TileType as parameter.

Make a switch case that returns a new base tile based on the type.

Reply:
Make it the responsibility of the calling function to cast it. Don’t cast it inside the function.

Reply:
I am not sure if I fully understand what you mean but I will give it shot. 

Should I return the correct tile based on the tile type?

Reply:
Done